packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1650,6 +1650,19 @@ class DropdownButtonFormField<T> extends FormField<T> {
            final InputDecoration effectiveDecoration = decorationArg.applyDefaults(
              Theme.of(field.context).inputDecorationTheme,
            );
+
+           final bool showSelectedItem = items != null && items.where((DropdownMenuItem<T> item) => item.value == state.value).isNotEmpty;
+           final bool showHintOrDisabledHint = () {
+             if (onChanged == null) {
+               return (hint != null || disabledHint != null) && !showSelectedItem;
+             } else if (items == null || items.isEmpty) {
+               return hint != null || disabledHint != null;
+             } else {
+               return hint != null && !showSelectedItem;
+             }
+           }();
+           final bool isEmpty = !showSelectedItem && !showHintOrDisabledHint;
```

Comment:
```suggestion
           final bool showSelectedItem = items != null && items.where((DropdownMenuItem<T> item) => item.value == state.value).isNotEmpty;
           bool showHintOrDisabledHint() {
             final bool isDropdownDisabled = onChanged == null || (items == null || items.isEmpty);
             if (isDropdownDisabled) {
               return hint != null || disabledHint != null;
             } else {
               return hint != null;
             }
           };
           final bool isEmpty = !showSelectedItem && !showHintOrDisabledHint();
```

What do you think about this logic? I think since we "and" `!showSelectedItem` when setting `isEmpty` we can move it out of `showHintOrDisabledHint`.

Reply:
If `onChange != null && hint != null && diabledHint == null && item == null`, the `isEmpty` should be true.

Reply:
If we wanna remove the `!showSelectedItem` logic, we should also change the variable name I think.

Reply:
@Renzo-Olivares you are right we can merge some logic together , how about change the method name to `showHintOrDisabledHintWhenNoSelectedItem`?